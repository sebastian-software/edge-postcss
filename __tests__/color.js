import { compile } from "./core"

// ====================================================
// ================== COLOR ===========================
// ====================================================

test("Improved Color Palette", () =>
  compile(".red { color: red; }")
)

test("RGBA with Hex", () =>
  compile(".red { color: rgba(#f11, 0.8); }")
)

test("Hex with Alpha", () =>
  compile(".blue { color: #11fc; }")
)

test("Color Function", () =>
  compile(".yellow { color: color(yellow a(90%)) }")
)

test("Easing Gradients", () =>
  compile(`
.demo {
  background: scrim-gradient(
    black,
    transparent
  );
}`))
