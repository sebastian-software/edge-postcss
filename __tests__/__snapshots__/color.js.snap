// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSS4 HSL Colors 1`] = `
".hsl{
color:red;
border-color:rgba(0,255,255,.2)
}"
`;

exports[`CSS4 HWB Colors 1`] = `
".hwb{
color:rgba(128,255,0,.5)
}"
`;

exports[`Color Function 1`] = `
".yellow{
color:rgba(255,220,0,.9)
}"
`;

exports[`Easing Gradients 1`] = `
".demo{
background:linear-gradient(#111 0%,rgba(17,17,17,.738) 19%,rgba(17,17,17,.541) 34%,rgba(17,17,17,.382) 47%,rgba(17,17,17,.278) 56.5%,rgba(17,17,17,.194) 65%,rgba(17,17,17,.126) 73%,rgba(17,17,17,.075) 80.2%,rgba(17,17,17,.042) 86.1%,rgba(17,17,17,.021) 91%,rgba(17,17,17,.008) 95.2%,rgba(17,17,17,.002) 98.2%,rgba(17,17,17,0) 100%)
}"
`;

exports[`Hex with Alpha 1`] = `
".blue{
color:rgba(17,17,255,.8)
}"
`;

exports[`Improved Color Palette 1`] = `
".red{
color:#ff4136
}"
`;

exports[`RGBA with Hex 1`] = `
".red{
color:rgba(255,17,17,.8)
}"
`;
